### ISSUE TITLE
Have a bot listening for releases on given repo and triggering a update applying a template via PR

### ISSUE THREAD

**Issue by tumido** [thoth-station/thoth-application#896]
**Is your feature request related to a problem? Please describe.**
Recently one of the projects we deploy (ODH) decided to do z-stream updates. On our end this update should be applied by a 2 line change in a manifest like this:

```diff
*** 25,28 ****
    repos:
      - name: manifests
!       uri: "https://github.com/opendatahub-io/odh-manifests/tarball/v1.0.0"
!   version: v1.0.0
--- 25,28 ----
    repos:
      - name: manifests
!       uri: "https://github.com/opendatahub-io/odh-manifests/tarball/v1.0.1"
!   version: v1.0.1
```

This change would have to happen on about 6 resource files and every week. Which is a repetitive task that we would like to offload to bots.

**Describe the solution you'd like**
In short "Patch a file based on a given template and open a PR".

It would be nice if we can specify what repositories the bot should listen to, and trigger on release in those repositories. The job would then choose a patch/template (defined in our repo, similarly to issue templates and such), apply this patch on a given resource file(s).

I imagine the end result to be a separate PR opened per file updated. No automerges.

**Describe alternatives you've considered**
Manual repetitive PRs. Opening about 6 of those 2 line PRs every week.

**Additional context**
Related to: https://github.com/operate-first/support/issues/89

**Comment by goern** [comment]
What repo is this? is it a tag on the repo? a new release on the repo? If we are receiving webhooks from opendatahub-io org it is fairly easy to add the action.

/sig cyborgs

**Comment by tumido** [comment]
In this particular case the event source would come from https://github.com/opendatahub-io/odh-manifests/releases as a release event. The PRs should be created against https://github.com/operate-first/apps/ repo content

**Comment by goern** [comment]
@harshad16 sounds like this is a tag-release pipeline with just the gitops tasks?

/assign @harshad16

**Comment by harshad16** [comment]
@tumido  apologies for losing track of this issue.
I see there is script already created https://github.com/operate-first/apps/blob/master/scripts/set-odh-release.sh
we have odh-manifest subscribed to aicoe-ci. so as @goern  suggested we can have a tag based trigger  check and run the script nad send out a pr to apps repo.
would this solution be something which would suffice for operate-first use case.

**Comment by tumido** [comment]
The solution you propose would be 100% sufficient. :+1: I think that's the best solution we could have hoped for :+1:

**Comment by harshad16** [comment]
Thanks for ack, will implement this and will display it to operate-first to get the feedback:+1:

**Comment by harshad16** [comment]
This seems to be working well for the operate-first.
closing the issue.
please let us know if any further features are required.
